### ISSUE TITLE
Differences with Backbone-Relational

### ISSUE THREAD

**Issue by philfreo** [dhruvaray/backbone-associations#32]
Just curious on all the known differences between backbone-associations and backbone-relational? I know that backbone-associations doesn't support reverse relations but I'm wondering what other functionality is different.

Would also be cool to see updated benchmarks since backbone-relational has had several big performance improvements recently.

**Comment by dhruvaray** [comment]
Sure... No worries. The section is now updated with the comparison figures against the latest Backbone-Relational library. I was waiting for the new tag (presumably 0.9.0 which officially declares support for Backbone 1.0.0). In the interim, I have used the latest commit SHA as a reference.

The primary motivation for starting off Backbone-associations was to provide a simpler (and faster) implementation of Backbone-Relation - sans the semaphores, event-queues and the like. More importantly, push as much as possible to the base Backbone framework methods. It therefore diverges in implementation fundamentally. 

It also does not provide any backing store for the fetched objects. We leave that to the application layer to manage.[Most SPA apps would have some kind of app-context container which would be a perfect store for shared objects and associations. They would be able to address the object identity problem better than a base framework - in my opinion] Consequently, we have left the fetch/save method open for application engineers to implement - as we have no backing store to worry about. 

Another goal has been to avoid ORM like connotations (and thereby constructs) in Backbone-associations - which typically come up when you try to map object graphs to relational table constructs (many-many relations mapped to an intermediate table and corresponding objects, polymorphic associations etc). 

[EDIT : Aug 20, 2013 : @sgrif, @monokrome jointly suggested and added a way to do polymorphic associations. Available from v0.5.1+]

Rather it focuses on regular containment relations found in traditional OO languages. In that sense, it is more "association" oriented and less "relational". We have had a pull request #12 - which infers reverse associations (in the sense of who-is-pointing-to-me) which does not need explicit syntax at model definition time. I am - at this point of time - inclined to pull it into the associations library. 

[EDIT : Aug 1, 2013 : Implicit reverse pointers are now part of v0.5.0]

Also sub-types are not supported at this point of time. But should we implement it, it would suffer from a derives-from type syndrome - rather than a true trait type. I suspect that to be the case in Backbone-relational as well.

Backbone-associations diverges from Backbone-relational when it comes to eventing. It changes the regular Backbone event names to a fully-qualified-event-path to the source of the update in the object graph. The rest of the event arguments match the regular Backbone arguments and are semantically consistent with respect to the source of update. Ditto for all the change-related methods provided by the base Backbone framework. 

Backbone-associations also handles cyclic object graphs and prevents infinite propagation of events for these graphs.

HTH

PS: Thanks for being one of the folks to star the project at it's inception!
